packages/apps/tools/src/components/Global/CustomTree/CustomTree.tsx

```diff
@@ -0,0 +1,224 @@
+                    onReload(item.data);
+                  }}
+                  className={reloadButtonStyles}
+                >
+                  <MonoCached
+                    color={token('color.icon.semantic.positive.default')}
+                  />
+                </Button>
+              ) : null}
+              {!item.data.isLoading && item.data.label ? (
+                <Badge size="sm" style={'highContrast'}>
+                  {item.data.label}
+                </Badge>
+              ) : null}
+              {item.data.isLoading ? (
+                <Badge size="sm" style={'highContrast'}>
+                  Loading
+                </Badge>
+              ) : null}
+            </Stack>
+            {item.isExpanded ? (
+              <Node
+                data={item.data.children}
+                level={1}
+                onItemClick={onItemClick}
+                onExpandCollapse={onExpandCollapse}
+              />
+            ) : null}
+          </>
+        );
+      }}
+    </CustomAccordion>
+  );
+}
+
+interface INodeProps<T>
+  extends Omit<ICustomAccordionProps<T>, 'data' | 'children'> {
+  data: TreeItem<T>[];
+  level: number;
+  onItemClick: (item: TreeItem<T>) => void;
+  onExpandCollapse: (item: TreeItem<T>, expanded: boolean) => void;
+}
+
+// eslint-disable-next-line react/function-component-definition
+function Node<T>({
+  data,
+  level,
+  onItemClick,
+  onExpandCollapse,
+  ...rest
+}: INodeProps<T>) {
+  return (
+    <CustomAccordion
+      {...rest}
+      data={data}
+      itemProps={{
+        style: {
+          borderBlockEnd:
+            level === 1 ? '1px solid rgba(0, 0, 0, 0.25)' : 'initial',
+        },
+      }}
+    >
+      {(child) => {
+        const hasChildren = !!child.data.children.length;
+        return (
+          <>
+            <Stack
+              alignItems={'center'}
+              justifyContent={'space-between'}
+              onClick={() => {
+                if (child.data.children.length) {
+                  child.toggleExpandCollapse();
+
+                  // We "toggle" the expanded state ourselves since this is still the previous state
+                  onExpandCollapse(child.data, !child.isExpanded);
+                } else {
+                  onItemClick(child.data);
+                }
+              }}
+              role="button"
+              className={itemContainerStyle}
+              style={{
+                paddingInlineStart: `${level * 20 + (!hasChildren ? 20 : 0)}px`,
+                cursor: hasChildren ? 'default' : 'pointer',
+              }}
+            >
+              {hasChildren ? (
+                <Button
+                  variant="transparent"
+                  onClick={() => {
+                    child.toggleExpandCollapse();
+
+                    // We "toggle" the expanded state ourselves since this is still the previous state
+                    onExpandCollapse(child.data, !child.isExpanded);
+                  }}
```

Comment:
Same as previous component, why a separate handler on the icon itself? If you don't want the icon to be clickable I would put pointerevents: none on the element and let the wrapper handle it.